app/models/user.rb

```diff
@@ -1,19 +1,21 @@
 class User < ApplicationRecord
   has_secure_password
 
-  before_save :downcase_nickname
+  before_validation :downcase_nickname
 
   validates :email,
-    presence: true,
-    uniqueness: true,
+    db_presence: true,
+    db_uniqueness: true,
     format: { with: /\A[a-z\d_+.\-]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i }
 
   validates :nickname,
-    presence: true,
-    uniqueness: true,
-    format: { with: /\A\w+\z/, message: " can contain only Latin letters, numbers, and a sign _" },
+    db_presence: true,
+    db_uniqueness: true,
+    format: { with: /\A\w+\z/ },
     length: { maximum: 40 }
 
+  private
+
   def downcase_nickname
     nickname.downcase!
```

Comment:
Imagine that there is no `nickname`, there will be 500 error. Read about [safe navigation](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/)